src/main/java/uk/gov/hmcts/reform/sscs/service/venue/VenueRpcDetailsService.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.reform.sscs.service.venue;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.model.VenueDetails;
+import uk.gov.hmcts.reform.sscs.service.VenueDataLoader;
+
+import java.util.Optional;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class VenueRpcDetailsService {
+
+    private final VenueDataLoader venueDataLoader;
+
+    public Optional<VenueRpcDetails> getVenue(String epimsId) {
+        return venueDataLoader.getVenueDetailsMap().values().stream().filter(this::isActiveVenue)
+            .map(VenueRpcDetails::new).filter(v -> v.getEpimsId().equalsIgnoreCase(epimsId)).findAny();
```

Comment:
I could live with the above, however it would be far easier to read if it were laid out like this:
return venueDataLoader.getVenueDetailsMap()
.values().stream()
.filter(this::isActiveVenue)
.map(VenueRpcDetails::new)
.filter(v -> v.getEpimsId().equalsIgnoreCase(epimsId))
.findAny();

Reply:
will do